NOTE-act-rules-common-aspects.bs

```diff
@@ -69,3 +69,18 @@ Language, either written or spoken, contained in nodes of the DOM [[DOM]] or acc
 Rules can only operate on the Language aspect if the language of a page can be determined. The means by which the language is assessed, whether by a person or a machine, is not of importance as long as the assessment meets the criteria defined in [[wcag2-tech-req#humantestable]] [[WCAG]].
 
 Some rules can only operate on a Language aspect if the language is sufficiently understood, while other rules only require identifying the language. For example, a rule checking that an accessible name is descriptive can only function if the language is understood. A rule checking the correctness of a `lang` attribute requires knowing what language is used, but not the meaning of the words.
+
+Source Code {#input-aspects-code}
+----------------------------------
+
+Source code, the programming language used to build a web page, may be of interest to ACT Rules that intend to test things like the presence of certain HTML elements or attributes. For example, an ACT Rule might test that elements do not contain duplicate attributes.
+
+Audio Output {#input-aspects-audio-out}
+---------------------------------------
+
+Audio output may be of interest to ACT Rules that intend to test things like the speech and sounds provided in media files. Some rules can operate on whether there is audio output, such as an ACT Rule that may check that audio output on a Web page does not play automatically. Some rules can only operate if the audio output is understood. For example, a rule checking that captions are equivalent can only function if the audio output is understood.
+
+Visual Output {#input-aspects-visual-out}
+-----------------------------------------
+
+Visual output may be of interest to ACT Rules that intend to test things like the moving images displayed in media or animation files. Some rules can operate on the presence of visual output, such as an ACT RUle that may check that visual output has a control to stop it. Some rules can only operate if the visual output is understood. For example, a rule checking that the visual output information is available in a transcript.
```

Comment:
Similar question to above about what understanding visual output means here.

Reply:
added understood "by the tester"